src/Microsoft.AspNetCore.Blazor.Browser.JS/src/Interop/InvokeJavaScriptFunctionWithJsonMarshalling.ts

```diff
@@ -0,0 +1,66 @@
+import { platform } from '../Environment';
+import { System_String } from '../Platform/Platform';
+import { getRegisteredFunction } from './RegisteredFunction';
+import { invokeDotNetMethod, MethodOptions, InvocationResult } from './InvokeDotNetMethodWithJsonMarshalling';
+import { getElementByCaptureId } from '../Rendering/ElementReferenceCapture';
+import { System } from 'typescript';
+import { error } from 'util';
+
+const elementRefKey = '_blazorElementRef'; // Keep in sync with ElementRef.cs
+
+export function invokeWithJsonMarshalling(identifier: System_String, ...argsJson: System_String[]) {
+  let result: InvocationResult;
+  const identifierJsString = platform.toJavaScriptString(identifier);
+  const args = argsJson.map(json => JSON.parse(platform.toJavaScriptString(json), jsonReviver));
+
+  try {
+    result = { succeeded: true, result: invokeWithJsonMarshallingCore(identifierJsString, ...args) };
+  } catch (e) {
+    result = { succeeded: false, message: e.message };
```

Comment:
We might need to be a bit more defensive about what we assume the `e` value is. Although it will usually be an `Error` instance (with `.message` and `.stack`), it doesn't necessarily have to be (e.g., if the target code does `throw someArbitraryObject` instead of `throw new Error('some message')`.

Perhaps we could have some logic like:
 * If `e instanceof Error` then set the message to `${e.message}\n${e.stack}`
 * Otherwise, set the message to `e ? e.toString() : null`